src/test/java/tests/UpdateServiceTests.java

```diff
@@ -0,0 +1,95 @@
+package tests;
+
+import static org.eclipse.egit.github.core.client.IGitHubConstants.CONTENT_TYPE_JSON;
+import static org.eclipse.egit.github.core.client.IGitHubConstants.SEGMENT_REPOS;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import github.GitHubClientExtended;
+import github.update.MilestoneUpdateService;
+import github.update.PageHeaderIterator;
+import github.update.UpdateService;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.egit.github.core.Milestone;
+import org.eclipse.egit.github.core.RepositoryId;
+import org.eclipse.egit.github.core.client.NoSuchPageException;
+import org.eclipse.egit.github.core.client.PagedRequest;
+import org.junit.Test;
+
+public class UpdateServiceTests {
+    /**
+     * Tests that head request to nonexistent repo throws an exception
+     * @throws IOException
+     */
+    @Test(expected = FileNotFoundException.class)
+    public void testHeadRequest() throws IOException {
+        GitHubClientExtended client = new GitHubClientExtended();
```

Comment:
These tests aren't authenticated, so could there be a problem with quota use? Or do HEAD requests not consume quota?